plugin/pkg/admission/security/podsecuritypolicy/admission.go

```diff
@@ -134,11 +138,7 @@ func (c *PodSecurityPolicyPlugin) Admit(a admission.Attributes) error {
 		if pod.ObjectMeta.Annotations == nil {
 			pod.ObjectMeta.Annotations = map[string]string{}
 		}
-		// set annotation to mark this as passed. Note, that the actual value is not important, the
-		// validating PSP might even change later-on. Also not that pspName can be the empty string
-		// if failOnNoPolicies is false.
-		// TODO: if failOnNoPolicies is toggled from false to true, we will never update the annotation anymore. Is this desired?
-		pod.ObjectMeta.Annotations[psputil.ValidatedPSPAnnotation] = pspName
+		pod.ObjectMeta.Annotations[psputil.InitiallyValidatedPSPAnnotation] = pspName
```

Comment:
note that before this PR we did not allow mutation in updates either (i.e. no serious mutation), but we were updating the annotation. Now we completely skip the mutating admission phase for update requests.